### PR TITLE
Add num workers cli arg

### LINKED ISSUE #6316: Dataloader number of workers in Trainer
https://github.com/huggingface/transformers/blob/175cd45e13b2e33d1efec9e2ac217cba99f6ae58/src/transformers/trainer.py#L252

If you want to use the Trainer from trainer.py, you only have the option to use only 0 number of workers for your dataloader.

However, even if I change the source code to have 10 number or workers for the data loader, the model still uses the same thread.

### PR DESCRIPTION
**Pull request by chadykamar**
Fixes #6316

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -122,6 +122,8 @@ class TrainingArguments:
         eval_steps (:obj:`int`, `optional`):
             Number of update steps between two evaluations if :obj:`evaluation_strategy="steps"`. Will default to the
             same value as :obj:`logging_steps` if not set.
+        dataloader_num_workers (:obj:`int`, `optional`, defaults to 0):
+            Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process.
```
Comment (sgugger) [inline src/transformers/training_args.py:126]:
```suggestion
            Number of subprocesses to use for data loading (PyTorch only). 0 means that the data will be loaded in the main process.
```

File: src/transformers/training_args.py [hunk 2]
```diff
@@ -259,6 +261,10 @@ class TrainingArguments:
         default=False, metadata={"help": "Drop the last incomplete batch if it is not divisible by the batch size."}
     )
     eval_steps: int = field(default=None, metadata={"help": "Run an evaluation every X steps."})
+    dataloader_num_workers: int = field(
+        default=0,
+        metadata={"help": "Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process."}
```
Comment (sgugger) [inline src/transformers/training_args.py:266]:
```suggestion
        metadata={"help": "Number of subprocesses to use for data loading (PyTorch only). 0 means that the data will be loaded in the main process."}
```

**Review by sgugger — APPROVED** [review]
Thanks for taking care of this. I just have two nits since the `TrainingArguments` are (for now) subclassed by TF so we need to specify it's just for PyTorch.

**Review by jplu — APPROVED** [review]
LGTM!

Just a tiny comment, this parameter might be useful for TF as well when doing some dataset pre-processing for the `num_parallel_calls` parameter of few functions such as [map](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#map) or [interleave](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#interleave)

**Comment by chadykamar** [comment]
I've addressed the comments regarding the docstring and help message. 

I'm a little less familiar with TensorFlow 2.0, but it seems like any preprocessing is done by the user before passing `train_dataset` and `eval_dataset` to the `TFTrainer` so there isn't an opportunity to set `num_parallel_calls` (I wasn't able to find any calls to `map` or `interleave` save for some markdown examples).

**Comment by jplu** [comment]
Indeed there is no call directly in the trainer and these functions have to be run directly in the example script. Nevertheless, this parameter is still useful as it can be used directly in the example script, instead of updating it manually.

**Pull request merged**